gate-keeper/readme.md

```diff
@@ -0,0 +1,105 @@
+## Introduction
+This guide will walk you through the process of building contracts for the gate keeper, deploying them into an existing world running in Docker, and testing their functionality by executing scripts.
+
+The gate keeper allows you to crowdfund items toward a specific goal. For example, if you need a certain amount of resources to build a new ship for the corporation etc.
+
+Previously, in the game gate keeper was used to have shared goals for players to contribute salt. Once the goals were met, gates and new areas were unlocked.
+
+## Deployment and Testing
+### Step 0: Deploy the gate keeper contracts to the existing world 
+First, copy the World Contract Address from the Docker logs obtained in the previous step, then run the following command:
+
+![alt text](../docker_deployment.png)
+
+```bash
+cd gate-keeper/packages/contracts
+```
+
+Install the dependecies for the contracts:
+```bash
+pnpm install
+```
+
+**Local Deployment**
+This will deploy the contracts to your local world.
+```bash
+pnpm run deploy:local --worldAddress <worldAddress> 
+```
+
+**Devnet Deployment**
+This will deploy the contracts to the Devnet world.
+```bash
+cd packages/contracts
+pnpm run deploy:devnet --worldAddress <worldAddress> 
+```
+
+
+eg: `pnpm run deploy:local --worldAddress 0xafc8e4fd5eee66590c93feebf526e1aa2e93c6c3`
+
+Once the deployment is successful, you'll see a screen similar to the one below. This process deploys the Gate Keeper contract.
+![alt text](./readme-imgs/deployment.png)
+
+
+### Step 1: Setup the environment variables 
```

Comment:
good to include info about where to obtain the .env file variables for the devnet environment

Reply:
Have sorted through all the readme's. Thank you :)